src/timer/timer_game_common.h

```diff
@@ -0,0 +1,292 @@
+		{}
+
+		bool operator < (const TPeriod &other) const
+		{
+			if (this->trigger != other.trigger) return this->trigger < other.trigger;
+			return this->priority < other.priority;
+		}
+
+		bool operator == (const TPeriod &other) const
+		{
+			return this->trigger == other.trigger && this->priority == other.priority;
+		}
+	};
+
+	using TElapsed = uint;
+	struct TStorage {};
+};
+
+/**
+ * Template class for time constants shared by both Calendar and Economy time.
+ */
+template <class T>
+class TimerGameConst {
+public:
+	static constexpr int DAYS_IN_YEAR = 365; ///< days per year
+	static constexpr int DAYS_IN_LEAP_YEAR = 366; ///< sometimes, you need one day more...
+	static constexpr int MONTHS_IN_YEAR = 12; ///< months per year
+
+	static constexpr int SECONDS_PER_DAY = 2;   ///< approximate seconds per day, not for precise calculations
+
+	/*
+	 * ORIGINAL_BASE_YEAR, ORIGINAL_MAX_YEAR and DAYS_TILL_ORIGINAL_BASE_YEAR are
+	 * primarily used for loading newgrf and savegame data and returning some
+	 * newgrf (callback) functions that were in the original (TTD) inherited
+	 * format, where 'TimerGame<T>::date == 0' meant that it was 1920-01-01.
+	 */
+
+	/** The minimum starting year/base year of the original TTD */
+	static constexpr typename TimerGame<T>::Year ORIGINAL_BASE_YEAR = 1920;
+	/** The original ending year */
+	static constexpr typename TimerGame<T>::Year ORIGINAL_END_YEAR = 2051;
+	/** The maximum year of the original TTD */
+	static constexpr typename TimerGame<T>::Year ORIGINAL_MAX_YEAR = 2090;
+
+	/** The absolute minimum & maximum years in OTTD */
+	static constexpr typename TimerGame<T>::Year MIN_YEAR = 0;
+
+	/** The default starting year */
+	static constexpr typename TimerGame<T>::Year DEF_START_YEAR = 1950;
+	/** The default scoring end year */
+	static constexpr typename TimerGame<T>::Year DEF_END_YEAR = ORIGINAL_END_YEAR - 1;
+
+	/**
+	 * MAX_YEAR, nicely rounded value of the number of years that can
+	 * be encoded in a single 32 bits date, about 2^31 / 366 years.
+	 */
+	static constexpr typename TimerGame<T>::Year MAX_YEAR = 5000000;
+
+	/** The date of the first day of the original base year. */
+	static constexpr typename TimerGame<T>::Date DAYS_TILL_ORIGINAL_BASE_YEAR = TimerGame<T>::DateAtStartOfYear(ORIGINAL_BASE_YEAR);
+
+	/** The date of the last day of the max year. */
+	static constexpr typename TimerGame<T>::Date MAX_DATE = TimerGame<T>::DateAtStartOfYear(MAX_YEAR + 1) - 1;
+
+	static constexpr typename TimerGame<T>::Year INVALID_YEAR = -1; ///< Representation of an invalid year
+	static constexpr typename TimerGame<T>::Date INVALID_DATE = -1; ///< Representation of an invalid date
+};
+
+/**
+ * Converts a Date to a Year, Month & Day.
+ * @param date the date to convert from
+ * @param ymd  the year, month and day to write to
+ */
+template <class T>
+/* static */ void TimerGame<T>::ConvertDateToYMD(TimerGame<T>::Date date, YearMonthDay *ymd)
```

Comment:
I came to realise this might be problematic, as I am not sure how often this is now generated in the code. But I also realised we can still move this to a .cpp file, it just needs a bit of help. I will thinking with it later today to give you an example.